src/storage/heap_file.c

```diff
@@ -21442,7 +21477,14 @@ heap_log_delete_physical (THREAD_ENTRY * thread_p, PAGE_PTR page_p, VFID * vfid_
   else
     {
       /* log record descriptor */
-      log_append_undoredo_recdes (thread_p, RVHF_DELETE, &log_addr, recdes_p, NULL);
+      if (thread_p->no_logging)
+	{
+	  log_append_undo_recdes (thread_p, RVHF_DELETE, &log_addr, recdes_p);
```

Comment:
What happens if the heap page was now flushed and failure occurs. In UNDO recovery phase, it seems for`heap_rv_undo_delete()`, which is the undo callback for `RVHF_DELETE`, to return an error (or duplicated records?) and it may cause logpb_fatal_error in log_rv_undo_record(). I haven't tested, just read and guess.

Reply:
@wrlawodms This code has been reverted. requests you to review for log records (mvcc) related to table records.